app/mqtt_reader.py

```diff
@@ -0,0 +1,86 @@
+"""MQTT Reader module.
+
+This module provides a client class for subscribing to topics and receiving messages from MQTT brokers.
+
+"""
+
+import logging
+from typing import Callable
+
+import paho.mqtt.client as mqtt
+
+from app.message import CommandMessage
+from app.exceptions import InvalidMessageError
+from app.error_handler import ErrorHandler
+from app.mqtt_client import MqttClient
+
+
+def _decode_message(message):
+    return message.payload.decode()
+
+
+class MqttReader(MqttClient):
+    _client: mqtt.Client
+
+    def __init__(
+        self,
+        port: int,
+        host: str,
+        error_handler: ErrorHandler,
+        client: mqtt.Client = None,
+    ) -> None:
+        super().__init__(host, port, client)
+        self.error_handler = error_handler
+        self.on_message_callbacks = []
+        self.topics = []
+
+    def subscribe_topics(self, topics: list[str]):
```

Comment:
as I think I said in a different place. I would rather make this method private and call it upon instantiation. I think you have everything when you 

Rather than doing this in main:

```
mqtt_client = setup_mqtt_client(configuration)

    mqtt_client.subscribe_topics([configuration.mqtt_settings.command_topic])
```
we should just do instantiation of the object and upon instantiation, we know where to connect.

Reply:
Good call. I removed this method altogether as it was confusingly named - it was just populating an internal attribute rather than actually doing the subscribe operation.